addons/mail/static/src/model/misc.js

```diff
@@ -56,3 +56,150 @@ export function isRelation(Model, fieldName) {
 export function isFieldDefinition(val) {
     return val?.[FIELD_DEFINITION_SYM];
 }
+
+export const fields = {
+    /**
+     * @template {keyof import("models").Models} M
+     * @param {M} targetModel
+     * @param {Object} [param1={}]
+     * @param {(this: Record) => any} [param1.compute] if set, the value of this relational field is declarative and
```

Comment:
We might want a typedef for the params common to all fields rather than having to re-define them

Reply:
Indeed, I didn't want to struggle with it for now, so I'll consider soon after